utils/src/copilot/copilot.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import * as vscode from "vscode";
+
+const languageModelPreference: { vendor: "copilot", family: string }[] = [
+    // not yet seen/available
+    // { vendor: "copilot", family: "gpt-4-turbo-preview" },
+    // seen/available
+    { vendor: "copilot", family: "gpt-4o" },
+    { vendor: "copilot", family: "gpt-4-turbo" },
+    { vendor: "copilot", family: "gpt-4" },
+    { vendor: "copilot", family: "gpt-3.5-turbo" },
+];
+
+async function selectMostPreferredLm(): Promise<vscode.LanguageModelChat | undefined> {
+    const lms = await vscode.lm.selectChatModels({ vendor: "copilot" });
+    const lmsInPreferredOrder = (lms || [])
+        .filter((lm) => languageModelPreference.some((pref) => pref.family === lm.family && pref.vendor === lm.vendor))
+        .sort((a, b) => languageModelPreference.findIndex((pref) => pref.family === a.family && pref.vendor === a.vendor) - languageModelPreference.findIndex((pref) => pref.family === b.family && pref.vendor === b.vendor));
+    return lmsInPreferredOrder?.at(0);
+}
+
+export function createPrimaryPromptToGetSingleQuickPickInput(picks: string[], relevantContext?: string): string {
```

Comment:
I'm not too concerned about how optimal this is yet. I think when we do this for other commands, we should probably have a way to pass custom prompts since it may not be a generic one shoe fits all scenario.